frontend/src/stimulus/controllers/dynamic/project-storage-form.controller.ts

```diff
@@ -209,13 +193,27 @@ export default class ProjectStorageFormController extends Controller {
       );
   }
 
-  private setProjectFolderModeQueryParam(mode:string) {
+  protected displayFolderSelectionOrLoginButton(isConnected:boolean, projectFolder:IStorageFile|null):void {
+    if (isConnected) {
+      this.selectProjectFolderButtonTarget.style.display = 'inline-block';
+      this.loginButtonTarget.style.display = 'none';
+      this.selectedFolderTextTarget.innerText = projectFolder === null
+        ? this.placeholderFolderNameValue
+        : projectFolder.name;
+    } else {
+      this.selectProjectFolderButtonTarget.style.display = 'none';
+      this.loginButtonTarget.style.display = 'inline-block';
+      this.selectedFolderTextTarget.innerText = this.notLoggedInValidationValue;
+    }
```

Comment:
Writing inline CSS in Javascript is rather bad style. I know that you only extracted that code but could you please refactor that to simply toggle a class that hides the elements?  e.g.:

to show an element: `this.selectProjectFolderButtonTarget.classList.remove('d-none');` 
to hide an element: `this.selectProjectFolderButtonTarget.classList.add('d-none');`